### PR TITLE
Supplier defined epics search bar

### PR DESCRIPTION
**Pull request by magnusjohn-bjss**

### CODE REVIEW

**Review by bjss-colin — DISMISSED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/SupplierDefinedEpicsController.cs [hunk 1]
```diff
@@ -133,5 +150,17 @@ public async Task<IActionResult> DeleteEpic(string epicId, DeleteSupplierDefined
 
             return RedirectToAction(nameof(Dashboard));
         }
+
+        private async Task<IEnumerable<Epic>> GetFilteredEpics(string search)
+        {
+            var allEpics = await supplierDefinedEpicsService.GetSupplierDefinedEpics();
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/SupplierDefinedEpicsController.cs:156]: why not supplierDefinedEpicsService.GetSupplierDefinedEpicsFiltered?

**Review by leerogers-bjss — DISMISSED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.Services/Capabilities/SupplierDefinedEpicsService.cs [hunk 2]
```diff
@@ -92,6 +92,20 @@ public Task<List<Epic>> GetSupplierDefinedEpics()
...
+            if (string.IsNullOrWhiteSpace(searchTerm))
+                throw new ArgumentNullException(nameof(searchTerm));
+
+            return dbContext.Epics
+                .AsNoTracking()
+                .Include(e => e.Capability)
+                .Where(e => e.SupplierDefined
+                    && (EF.Functions.Like(e.Name, $"%{searchTerm}%")
+                        || EF.Functions.Like(e.Capability.Name, $"%{searchTerm}%")))
+                .ToListAsync();
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.Services/Capabilities/SupplierDefinedEpicsService.cs:106]: please add .Take(15) to limit to 15 only. thanks

**Review by leerogers-bjss — CHANGES_REQUESTED** [review]
just a minor change then will approve

File: src/NHSD.GPIT.BuyingCatalogue.Services/Capabilities/SupplierDefinedEpicsService.cs (hunk 2 above)
Comment (magnusjohn-bjss) [reply src/NHSD.GPIT.BuyingCatalogue.Services/Capabilities/SupplierDefinedEpicsService.cs:106]: added in controller

**Review by leerogers-bjss — APPROVED** [review]

**Review by Okechukwu802 — APPROVED** [review]

**Pull request merged**